src/drivers/net/mod.rs

```diff
@@ -25,11 +24,9 @@ const POLL_PERIOD: u64 = 20_000;
 fn set_polling_mode(value: bool) {
 	// is the driver already in polling mode?
 	if POLLING.swap(value, Ordering::SeqCst) != value {
-		irqsave(|| {
-			if let Some(driver) = crate::arch::kernel::pci::get_network_driver() {
-				driver.borrow_mut().set_polling_mode(value);
-			}
-		});
+		if let Some(driver) = crate::arch::kernel::pci::get_network_driver() {
+			driver.lock().set_polling_mode(value);
```

Comment:
This might deadlock. `virtio_irqhandler` also gets the network driver and calls
```rust
driver.lock().handle_interrupt()
```

The interrupt might happen while the driver is locked here.

Reply:
That is the reason for SpinlockIrqSave. If a task hold this lock, all interrupt are blocked. => No interrupt during the critical section

Reply:
You find more details at https://github.com/hermitcore/libhermit-rs/blob/master/src/synch/spinlock.rs#L212-L221